package.json

```diff
@@ -195,16 +191,20 @@
     "@types/url-parse": "^1.4.3",
     "@types/uuid": "^3.4.4",
     "@types/validator": "^9.4.2",
-    "@typescript-eslint/parser": "^3.10.1",
+    "@typescript-eslint/eslint-plugin": "^3.10.1",
+    "@typescript-eslint/parser": "^4.28.4",
     "abortcontroller-polyfill": "1.3.0",
     "babel-jest": "^25.1.0",
     "babel-preset-react-native": "^4.0.1",
     "chalk": "^2.4.1",
     "danger": "^10.3.0",
     "danger-plugin-digitalcitizenship": "^0.3.1",
-    "eslint": "^6.5.1",
-    "eslint-plugin-import": "^2.22.0",
-    "eslint-plugin-react": "^7.20.6",
+    "eslint": "^7.31.0",
+    "eslint-config-prettier": "^8.3.0",
+    "eslint-plugin-functional": "^3.2.1",
+    "eslint-plugin-import": "^2.23.4",
+    "eslint-plugin-react": "^7.24.0",
+    "eslint-plugin-sonarjs": "^0.5.0",
```

Comment:
In this case could be possible to upgrade to `0.9.1` but the update here is a bit tricky because there are some new rules (`sonarjs/no-nested-template-literals, sonarjs/no-ignored-return, sonarjs/no-nested-switch and sonarjs/no-empty-collection`) that requires some code refactoring. What you think about that, could be better to update the library now and refactor or do it in a second iteration?

Reply:
it's just 20 errors and many of them look trivial, but there is something spooky like [this export](https://github.com/pagopa/io-app/blob/ac6e9ef811711a60d8f6cc5733051bb5c2b07788/ts/utils/paymentMethodCapabilities.ts#L6) which is not used anywhere in the app.

Since it requires some more work but is doable, I'm coming back to this once I'm done with other stuff, likely tomorrow afternoon.

Reply:
I've addressed most of the errors, though:
* I disabled locally [no-empty-collection](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-empty-collection.md) in some cases, since it looks like they are there to accommodate specific use cases
* I turned [no-nested-template-literals](https://github.com/SonarSource/eslint-plugin-sonarjs/blob/master/docs/rules/no-nested-template-literals.md) to warning because it affects readability only and requires some refactoring